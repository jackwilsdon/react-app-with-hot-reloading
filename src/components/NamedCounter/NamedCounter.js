import Counter from '../Counter';
import PropTypes from 'prop-types';
import React from 'react';

const NamedCounter = ({ name, ...props }) => (
  <div className="NamedCounter">
    {name && <label className="NamedCounter__label">{name}</label>}
    <div className="NamedCounter__counter">
      <Counter {...props} />
    </div>
  </div>
);

NamedCounter.propTypes = {
  name: PropTypes.string,
};

NamedCounter.defaultProps = {
  name: null,
};

export default NamedCounter;
