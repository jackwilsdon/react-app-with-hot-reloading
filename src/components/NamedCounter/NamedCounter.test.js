import NamedCounter from './NamedCounter';
import React from 'react';
import noop from 'lodash/noop';
import { shallow } from 'enzyme';

it('renders correctly', () => {
  const tree = shallow(
    <NamedCounter
      name="Counter name"
      onAdd={noop}
      onSubtract={noop}
      value={12345}
    />,
  );

  expect(tree).toMatchSnapshot();
});
