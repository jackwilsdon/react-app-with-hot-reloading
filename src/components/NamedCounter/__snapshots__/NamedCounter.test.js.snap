// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="NamedCounter"
>
  <label
    className="NamedCounter__label"
  >
    Counter name
  </label>
  <div
    className="NamedCounter__counter"
  >
    <Connect(Counter)
      identifier="default"
      onAdd={[Function]}
      onSubtract={[Function]}
      value={12345}
    />
  </div>
</div>
`;
